File: examples/cmdline1/python/main.py
import argparse
import sys

def main():
    # REQ-007: Uses Python programming language
    # REQ-006: Single source file
    parser = argparse.ArgumentParser(description="Description: show this help dialog")
    
    # -n --name
    parser.add_argument("-n", "--name", type=str, help="Description: provide a name to greet")
    
    # -r --repeat (N:int)
    parser.add_argument("-r", "--repeat", type=int, help="Description: repeat word N times.")

    args = parser.parse_args()

    # Logic to handle name and repeat
    word_to_print = f"Hello, {args.name}" if args.name else "Hello"
    
    count = args.repeat if args.repeat is not None else 1

    # If no arguments provided (other than help implicit), show help
    if len(sys.argv) == 1:
        parser.print_help()
        sys.exit(1)

    for _ in range(count):
        print(word_to_print)

if __name__ == "__main__":
    main()

File: examples/cmdline1/go/main.go
package main

import (
	"flag"
	"fmt"
	"os"
)

func main() {
	// Define flags
	namePtr := flag.String("name", "", "Description: provide a name to greet")
	shortNamePtr := flag.String("n", "", "Description: provide a name to greet (shorthand)")
	
	repeatPtr := flag.Int("repeat", 1, "Description: repeat word N times.")
	shortRepeatPtr := flag.Int("r", 1, "Description: repeat word N times. (shorthand)")

	// Custom help description
	flag.Usage = func() {
		fmt.Fprintf(os.Stderr, "Usage of %s:\n", os.Args[0])
		fmt.Fprintf(os.Stderr, "  -h, --help      Description: show this help dialog\n")
		fmt.Fprintf(os.Stderr, "  -n, --name      Description: provide a name to greet\n")
		fmt.Fprintf(os.Stderr, "  -r, --repeat    Description: repeat word N times.\n")
	}

	flag.Parse()

	// Consolidate shorthand and long-form flags
	name := *namePtr
	if name == "" {
		name = *shortNamePtr
	}

	repeat := *repeatPtr
	if repeat == 1 && *shortRepeatPtr != 1 {
		repeat = *shortRepeatPtr
	}

	// Logic execution
	greeting := "Hello"
	if name != "" {
		greeting = fmt.Sprintf("Hello, %s", name)
	}

	for i := 0; i < repeat; i++ {
		fmt.Println(greeting)
	}
}
